packages/utilities/src/useFocusRects.ts

```diff
@@ -151,16 +173,16 @@ function _onPointerDown(ev: PointerEvent, providerElem?: Element): void {
 // every element that is being tabbed into.
 // This works because `classList.add` is smart and will not duplicate a classname that already exists on the classList
 // when focus visibility is turned on.
-function _onKeyDown(ev: KeyboardEvent, providerElem?: Element): void {
+function _onKeyDown(ev: KeyboardEvent, registeredProviders?: HTMLElement[]): void {
```

Comment:
Are there any cases where these handlers could run for multiple elements where it potentially sets the visibility multiple times on the same provider element? Not sure if that can happen, but the potential scenario came to mind

Reply:
I don't know of a case where the same provider could be added multiple times, but it's not a problem if it does. In fact, this code was already updating the classes multiple times in a row before my change, since it would add the class on every keystroke. It turns out not to be a problem because `HTMLElement.classList.add()` ignores duplicate classes. So adding it multiple times is not a concern.